downloader/src/funTest/kotlin/DownloaderTest.kt

```diff
@@ -180,5 +180,37 @@ class DownloaderTest : StringSpec() {
             tyrexDir.isDirectory shouldBe true
             tyrexDir.walk().count() shouldBe 409
         }
+
+        "Can download a ZIP source artifact from GitHub".config(tags = setOf(ExpensiveTag)) {
+            val url = "https://github.com/microsoft/tslib/archive/1.10.0.zip"
+            val pkg = Package(
+                id = Identifier(
+                    type = "NPM",
+                    namespace = "",
+                    name = "tslib",
+                    version = "1.10.0"
+                ),
+                declaredLicenses = sortedSetOf(),
+                description = "",
+                homepageUrl = "",
+                binaryArtifact = RemoteArtifact.EMPTY,
+                sourceArtifact = RemoteArtifact(
+                    url = url,
+                    hash = Hash.create("7f7994408f130dd138a59a625eeef3be1ab40f7b")
+                ),
+                vcs = VcsInfo.EMPTY
+            )
+
+            val downloadResult = Downloader.download(pkg, outputDir)
+            downloadResult.vcsInfo shouldBe null
+            downloadResult.sourceArtifact shouldNotBe null
```

Comment:
Line 206 is redundant with below `!!` and can be removed.

Reply:
That's basically true, but I simply copied this from the test case above, which uses @mnonnenmacher's style of explicitly expressing that "sourceArtifact should not be null". We use that explicit style in most of tests in the code base, I believe.

Reply:
Why not pick the _better_ solution as consistency is absent anyways?

Reply:
Because what's "better" apparently is subjective. As I recall, @mnonnenmacher prefers to have these explicit assertions, and I don't really care so much in this case, so we simply kept it as-is until now. If you feel like this should be changed, you should try to convince @mnonnenmacher, not myself.

Reply:
BTW, I'm again linking to https://github.com/kotest/kotest/issues/986, which would be the "best" solution, if it was possible to implement in Kotlin.

Reply:
From the options mentioned in that ticket I would prefer `shouldNotBeNull()` over using `!!` in the next line.

Reply:
> Because what's "better" apparently is subjective. 

I didn't mean to make a statement which of the options is better, I was just confused by the argumentation that "it should be like this because it was like this before".  I'm fine with just leaving this as it was before.

Reply:
But actually the `shouldNotBeNull()` approach looks very attractive to me, as it makes use of Kotlin contracts exactly how they are supposed to be used. In fact, any solutions that gets rid of a "hard-cast" via `!!` is attractive to me. I'd be happy with migrating our test code base to the `shouldNotBeNull()` approach.

Reply:
@sschuberth See: https://github.com/oss-review-toolkit/ort/pull/2870